### DISCUSSION TITLE
Talk about Slot migration for BloomFilter

### DISCUSSION THREAD

**Discussion by zncleon** [apache/kvrocks#1775] [General]
Currently, kvrocks performs slot migration by converting every batch data to RESP command and then sending it to the destination. But I found it's hard to extract RESP about `bf.madd` from batch for two reason:

1.  The inserted items would not be included in both `key` and `value`, and we can't get items from value(like bitmap), it's a irreversible conversion.
2.  I also can't assign items to log's `args` in advance for we can't use different args[i] in different batches.
---
I find two way to resolve it:

1.  Use slot migration based on raw key value #1223, but the feature seems not completed and I don't know the details about it.
2. To support some commands like bf.loadchunk, and `bf.madd` batch would be converted to this RESP command. The batch's value as the command data.

Which way would work, or have some other ways? 

---
[1] [https://redis.io/commands/bf.loadchunk/](https://redis.io/commands/bf.loadchunk/)

**Comment by git-hulk** [comment]
I prefer the first solution, but it may need some time to test if this PR works well.
